components/script/dom/htmldialogelement.rs

```diff
@@ -124,4 +125,36 @@ impl HTMLDialogElementMethods<crate::DomTypeHolder> for HTMLDialogElement {
             .dom_manipulation_task_source()
             .queue_simple_event(target, atom!("close"));
     }
+
+    /// <https://html.spec.whatwg.org/multipage/#dom-dialog-requestclose>
+    fn RequestClose(&self, return_value: Option<DOMString>, can_gc: CanGc) {
+        let element = self.upcast::<Element>();
+        let target = self.upcast::<EventTarget>();
+
+        // 1. If this does not have an open attribute, then return.
+        if !element.has_attribute(&local_name!("open")) {
+            return;
+        }
+
+        // TODO: 2. Assert: this's close watcher is not null.
+        // 3. Set dialog's enable close watcher for requestClose() to true.
+        // 4. If returnValue is not given, then set it to null.
+        // 5. Set this's request close return value to returnValue.
+        // 6. Request to close dialog's close watcher with false.
+        // 7. Set dialog's enable close watcher for requestClose() to false.
+        // ADHOC: This implementation doesn't follow the steps above, but has the same observable behavior.
```

Comment:
In general I think not following the spec requires a good reason(which I don't see here), but in this particular case, given the explanation for [`close-watcher-manager`](https://html.spec.whatwg.org/multipage/interaction.html#close-watcher-manager) is "anti-abuse protections designed to prevent developers from disabling users' history traversal abilities", I am against implementing this method without the proper backing infrastructure.

Reply:
This method would "bypass" that anti abuse mechanism per the spec. So this doesn't introduce anything bad in that way. That anti abuse mechanism is designed for Android back gestures as close Requests to close watchers.

This is a JS method and so doesn't have the same issue of being used for multiple things.

Reply:
To be specific:

Step 6: "[Request to close](https://html.spec.whatwg.org/multipage/interaction.html#close-watcher-request-close) dialog's [close watcher](https://html.spec.whatwg.org/multipage/interactive-elements.html#dialog-close-watcher) with false." - the false here, goes through and is used in step 6 of the [close watch request to close algorithm](https://html.spec.whatwg.org/multipage/interaction.html#close-watcher-request-close).

Where the false value bypasses the history-action user activation checks (the core of the anti abuse mechanics).

Reply:
Having said that I did also get some draft PR for close watchers but ran into some weird issues regarding Pointer types so perhaps I can push that up and see if we can work it out. Just thought I'd point out that this isn't an actually bad bypass to anything.